packages/grid/x-data-grid-pro/src/utils/tree/getNonAutoGeneratedDescendants.ts

```diff
@@ -0,0 +1,20 @@
+import { GridRowTreeConfig, GridRowId } from '@mui/x-data-grid';
+
+export const getNonAutoGeneratedDescendants = (tree: GridRowTreeConfig, parentId: GridRowId) => {
+  const children = tree[parentId]?.children;
+  if (children == null) {
+    return [];
+  }
+
+  const validDescendants: GridRowId[] = [];
+  for (let i = 0; i < children.length; i += 1) {
+    const child = children[i];
+    const childNode = tree[child];
+    if (!childNode.isAutoGenerated) {
+      validDescendants.push(child);
+    }
+    validDescendants.push(...getNonAutoGeneratedDescendants(tree, childNode.id));
```

Comment:
I have a doubt that this method returns rows in the order provided to the DataGrid. For example


For example in the following tree, since you use a recursive function to navigate in the tree, the id 1 must be next to the id 2. And the same for ids 3 and 4
A
  - B
    - 1
    - 2
  - C
    - 3
    - 4


But the ids can be provides as follow:
```js
rows=[
	{id: 1, path="A/B"},
	{id: 3, path="A/C"},
	{id: 4, path="A/C"},
	{id: 2, path="A/B"},
]
```

I think the best solution is to change the dos to say that if apply sorting is `false` the order is not specified but perf are better